platform-sdk/swirlds-benchmarks/src/jmh/java/com/swirlds/benchmark/reconnect/MerkleBenchmarkUtils.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.benchmark.reconnect;
+
+import static com.swirlds.common.merkle.copy.MerkleInitialize.initializeTreeAfterCopy;
+import static com.swirlds.common.threading.manager.AdHocThreadManager.getStaticThreadManager;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.merkle.MerkleInternal;
+import com.swirlds.common.merkle.MerkleLeaf;
+import com.swirlds.common.merkle.MerkleNode;
+import com.swirlds.common.merkle.crypto.MerkleCryptoFactory;
+import com.swirlds.common.merkle.synchronization.LearningSynchronizer;
+import com.swirlds.common.merkle.synchronization.TeachingSynchronizer;
+import com.swirlds.common.merkle.synchronization.config.ReconnectConfig;
+import com.swirlds.common.merkle.synchronization.utility.MerkleSynchronizationException;
+import com.swirlds.common.threading.pool.StandardWorkGroup;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.platform.network.SocketConfig;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+/**
+ * A utility class to support benchmarks for reconnect.
+ */
+public class MerkleBenchmarkUtils {
+    /**
+     * Returns the following tree:
+     *
+     * <pre>
+     *             root
+     *           / |   \
+     *          A  i0  i1
+     *             /\  /\
+     *            B C D null
+     * </pre>
+     */
+    static BenchmarkMerkleInternal buildLessSimpleTree() {
```

Comment:
This method is used only in `ReconnectRunner`, and only to replace one of the (grand)children with a virtual map. What if this method accepts a `VirtualMap` instance and returns the simplest tree possible: a root (merkle internal) with a single leaf (virtual map)?